ibtk/include/ibtk/libmesh_utilities.h

```diff
@@ -975,6 +975,104 @@ get_nodal_dof_indices(const libMesh::DofMap &dof_map,
 #endif
 }
 
+
+/**
+ * Return the maximum edge length of a given element with mapped nodes.
+ *
+ * @p elem The given libMesh element.
+ * @p X_node Mapped coordinates of the nodes.
+ */
+template <class MultiArray>
+inline double
+get_max_edge_length(const libMesh::Elem* const elem, const MultiArray& X_node)
+{
+#ifndef NDEBUG
+    TBOX_ASSERT(elem->n_nodes() == X_node.shape()[0]);
+    TBOX_ASSERT(elem->dim() == X_node.shape()[1]);
+#endif
+    const libMesh::ElemType elem_type = elem->type();
+
+    double max_edge_length_2 = 0.0;
+    // TODO: implement additional element-specific versions as needed.
+    switch (elem_type)
+    {
+    case libMesh::TET4:
+    {
+        // all nodes are connected to all other nodes
+        constexpr unsigned int edge_pairs[6][2] = { { 0, 1 }, { 0, 2 }, { 0, 3 }, { 1, 2 }, { 1, 3 }, { 2, 3 } };
```

Comment:
Is there any way to look this up from `libMesh` rather than writing it out by hand?

Reply:
AFAICT no: this table is only available in the source file.